### ISSUE TITLE
EventBridge event-pattern not correctly matched on detail content

### ISSUE THREAD

**Issue by jesmiatka** [localstack/localstack#3369]
# Type of request: This is a ...

[x] bug report
[ ] feature request

# Detailed description

It looks like the event pattern of EventBridge is not applied correctly on the `detail` field.

## Expected behavior

The provided example does not put a message in SQS.

## Actual behavior

The provided example puts a message in SQS.

# Steps to reproduce

## Command used to start LocalStack

The following scripts returns an item in SQS, but should not. The bit `detail: EventType` is not matched. Only values "0" and "1" should be pushed, whilst we put the value in the event "2".

```
awslocal events create-event-bus \
    --name detail-bus-1

awslocal sqs create-queue \
    --queue-name "detail-sqs-1"

awslocal events put-rule \
    --name "detail-rule-1" \
    --event-bus-name "detail-bus-1" \
    --event-pattern '{ "detail": { "EventType": [ "0", "1" ] } }'

awslocal events put-targets \
    --rule "detail-rule-1" \
    --event-bus-name "detail-bus-1" \
    --targets "Id"="detail-target-rule-1","Arn"="arn:aws:sqs:us-east-1:000000000000:detail-sqs-1","InputPath"="$.detail"

awslocal events put-events --entries '[{"DetailType": "test", "Source": "aws.test", "Detail": "{ \"EventType\": \"2\" }", "EventBusName": "detail-bus-1"}]'
awslocal sqs receive-message --queue-url http://localhost:4566/000000000000/detail-sqs-1
```

**Comment by whummer** [comment]
Thanks for reporting @jesmiatka . This should be fixed in #3922 - can you please pull the `latest` Docker image and give it another try? Please let us know if the problem persists. Thanks!

**Comment by mtossy** [comment]
I've noticed that the issue regarding the missing '=' operator in event pattern matching logic is still prevalent, as demonstrated in the documentation example `"Price": [{ "numeric": ["=", 100] }]`. It seems that the corresponding logic for the '=' operator is not implemented in the `handle_numeric_conditions` function.
https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-event-patterns.html

This could lead to unexpected behavior when users attempt to match events with numeric conditions that are supposed to check for equality.

I'm encountering this issue on [the latest release/version number], and it seems to be directly related to the functionality added in PR #3922. Could this be looked into?
